### PR TITLE
gh-118131: Command-line interface for the `random` module

### PR DESCRIPTION
**Pull request by hugovk**
* Issue: gh-118131

----
📚 Documentation preview 📚: https://cpython-previews--118132.org.readthedocs.build/en/118132/library/random.html#command-line-usage

### CODE REVIEW

**Review by pfmoore — APPROVED** [review]

**Comment by rhettinger** [comment]
Personally, I don't support this proposal but won't stand in the way of it either.  It seems gratuitous.

File: Lib/random.py [hunk 1]
```diff
@@ -996,5 +996,75 @@ def _test(N=10_000):
...
+        return uniform(1, args.float)
+
+    if args.test:
+        _test(args.test)
+        return ""
+
+    # No explicit argument, select based on input
+    if len(args.input) == 1:
+        val = args.input[0]
+        try:
```
Comment (funkyfuture) [inline Lib/random.py:1050]: what about `str.digit()`?

Comment (hugovk) [reply Lib/random.py:1050]: We need to turn the string into an int anyway, and there are some things where `str.isdigit()` is true but we can't `int()` them:

```pycon
>>> "²".isdigit()
True
>>> int("²")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '²'
```

**Review by erlend-aasland — APPROVED** [review]

File: Lib/random.py [hunk 2]
```diff
@@ -996,5 +996,75 @@ def _test(N=10_000):
     _os.register_at_fork(after_in_child=_inst.seed)
 
 
+# ------------------------------------------------------
+# -------------- command-line interface ----------------
+
+
+def parse_args(arg_list: list[str] | None):
```
Comment (layday) [inline Lib/random.py]: Might be better to make this private?

```suggestion
def _parse_args(arg_list: list[str] | None):
```

File: Lib/random.py [hunk 3]
```diff
@@ -996,5 +996,75 @@ def _test(N=10_000):
...
+
+    # Explicit arguments
+    if args.choice:
+        return choice(args.choice)
+
+    if args.integer is not None:
+        return randint(1, args.integer)
+
+    if args.float is not None:
+        return uniform(1, args.float)
```
Comment (andersk) [inline Lib/random.py:1041]: In #118131 this was specced for floats to use the range between `0` and `N` of length `N`, which would be more useful and makes a lot more sense than the range between `1` and `N` of length `N - 1`.

- #123968
- #123971

**Pull request merged**